spark/src/main/scala/geotrellis/spark/ingest/Tiler.scala

```diff
@@ -1,78 +1,66 @@
-   * @param createKey function to create target key, K, given corresponding source key, T, and a [[SpatialComponent]]
-   * @param rdd       rdd containing source tiles, may be overlapping
-   * @param metaData  RasterMetaData result tiles
-   * @tparam T        type of input tile key
-   * @tparam K        type of output tile key, must have SpatialComponent
-   * @return          RasterRDD[K] of mosaicked tiles
-   */
-  def apply[T, K: SpatialComponent: ClassTag]
-    (getExtent: T=> Extent, createKey: (T, SpatialKey) => K)
-    (rdd: RDD[(T, Tile)], metaData: RasterMetaData)
-  : RasterRDD[K] = {
-
-    val bcMetaData = rdd.sparkContext.broadcast(metaData)
-
-    val tilesWithKeys: RDD[(K, (K, Extent, Tile))] =
-      rdd
-        .flatMap { { tup =>
+  def cutTiles[T, K: SpatialComponent: ClassTag] (
+    getExtent: T=> Extent,
+    createKey: (T, SpatialKey) => K,
+    rdd: RDD[(T, Tile)],
+    mapTransform: MapKeyTransform,
+    cellType: CellType,
+    tileLayout: TileLayout
+  ): RDD[(K, Tile)] =
+    rdd
+      .flatMap { tup =>
         val (inKey, tile) = tup
-        val metaData = bcMetaData.value
         val extent = getExtent(inKey)
-        metaData.mapTransform(extent)
+        mapTransform(extent)
           .coords
           .map  { spatialComponent =>
-          val outKey = createKey(inKey, spatialComponent)
-          (outKey, (outKey, extent, tile))
-        } }
-      }
-
-    // Functions for combine step
-    val createTile =  { (tup: (K, Extent, Tile)) => {
-      val (key, extent, tile) = tup
-      //val metaData = bcMetaData.value
-      val tmsTile = ArrayTile.empty(metaData.cellType, metaData.tileLayout.tileCols, metaData.tileLayout.tileRows)
-      tmsTile.merge(metaData.mapTransform(key), extent, tile)
-    }}
+            val outKey = createKey(inKey, spatialComponent)
+            val tmsTile = ArrayTile.empty(cellType, tileLayout.tileCols, tileLayout.tileRows)
+            tmsTile.merge(mapTransform(outKey), extent, tile)
 
-    val combineTiles1 =  { (tile: MutableArrayTile, tup: (K, Extent, Tile)) => {
-      val (key, extent, prevTile) = tup
-      val metaData = bcMetaData.value
-      tile.merge(metaData.mapTransform(key), extent, prevTile)
-    }}
+            (outKey, tmsTile)
+          }
+       }
 
-    val combineTiles2 = { (tile1: MutableArrayTile, tile2: MutableArrayTile) =>
-      tile1.merge(tile2)
-    }
-
-    val newRdd: RDD[(K, Tile)] =
-      new PairRDDFunctions(tilesWithKeys)
-        .combineByKey(createTile, combineTiles1, combineTiles2)
-        .map { case (key, tile) => (key, tile: Tile) }
+  def apply[T, K: SpatialComponent: ClassTag](
+    getExtent: T=> Extent,
+    createKey: (T, SpatialKey) => K,
+    rdd: RDD[(T, Tile)],
+    mapTransform: MapKeyTransform,
+    cellType: CellType,
+    tileLayout: TileLayout
+  ): RDD[(K, Tile)] =
+    cutTiles(getExtent, createKey, rdd, mapTransform, cellType, tileLayout)
+      .reduceByKey { case (tile1: Tile, tile2: Tile) =>
```

Comment:
The combiners in combineByKey can be re-used, it's safe to return the reference to leftmost argument to mergeCombiners. Using reduceByKey requires allocation of a new object on every merge. Is it enough to make a difference?  I guess it depends on how many layout tiles the source tile intersects with.